xbmc/Util.cpp

```diff
@@ -2467,3 +2468,19 @@ void CUtil::CopyUserDataIfNeeded(const std::string& strPath,
     CFile::Copy(srcPath, destPath);
   }
 }
+
+bool CUtil::FileDeleteWithConfirm(const CFileItemPtr& item)
```

Comment:
Replace `CFileItemPtr` with `std::shared_ptr>CFileItem>`.

Reply:
Didn´t we have a GUI utils or something similar @ksooo? I would avoid adding this to this generic utils class

Reply:
We have `CFileUtils` which already has GUI deps and I already proposed to move this code there.

Reply:
```c++
bool CFileUtils::RenameFile(const std::string &strFile)
{
  std::string strFileAndPath(strFile);
  ...
  if (CGUIKeyboardFactory::ShowAndGetInput(strFileName, CVariant{g_localizeStrings.Get(16013)}, false))
```

Reply:
So, having `CFileUtils::DeleteFile` would fit quite nicely.

Reply:
That's what I tried at first but it didn't compile.